test/common/http/http2/http2_frame_test.cc

```diff
@@ -0,0 +1,37 @@
+#include <string>
+
+#include "envoy/http/metadata_interface.h"
+
+#include "common/http/http2/metadata_encoder.h"
+
+#include "test/common/http/http2/http2_frame.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+class EqualityMetadataFrame : public ::testing::Test {};
+
+// From metadata map
+TEST(EqualityMetadataFrame, Http2FrameTest) {
+  MetadataMap metadataMap = {{"Connections", "15"}, {"Timeout Seconds", "10"}};
+  Http2Frame http2FrameFromUtility = Http2Frame::makeMetadataFrameFromMetadataMap(
+      1, metadataMap, Http2Frame::MetadataFlags::EndMetadata);
+  MetadataEncoder metadataEncoder{};
+  Http::MetadataMapPtr metadataMapPtr = std::make_unique<Http::MetadataMap>(metadataMap);
+  Http::MetadataMapVector metadata_map_vector;
+  metadata_map_vector.push_back(std::move(metadataMapPtr));
+  metadataEncoder.createPayload(metadata_map_vector);
+  std::string payloadFromEncoder = metadataEncoder.payload();
+  std::string payloadFromHttp2Frame(http2FrameFromUtility);
+  // Note: the actual encoding of the metadata map is nondeterministic and flaky. This is okay!
+  ASSERT_EQ(static_cast<int>(http2FrameFromUtility.type()), 0x4D); // type
+  ASSERT_EQ(payloadFromHttp2Frame[4], 4);                          // flags
+  ASSERT_EQ(std::to_string(payloadFromHttp2Frame[8]),
```

Comment:
I wonder if you can add this test to metadata_encoder_decoder_test.cc

Like this, but you manually set `output_buffer_` with your frame instead of invoking it through `submitMetadata`. Do you think that works?

Reply:
Sorry, don't really understand what you mean by this.

Reply:
Moved test, added logic.